app/src/main/java/com/example/househomey/SignInFragment.java

```diff
@@ -0,0 +1,200 @@
+                                                activity.startActivity(intent);
+                                            }
+                                        }
+                                    } else {
+                                        usernameEdittext.setError("username or password not recognized");
+                                        passwordEdittext.setError("username or password not recognized");
+
+                                    }
+                                });
+                            }
+                        } else {
+                            usernameEdittext.setError("username or password not recognized");
+                            passwordEdittext.setError("username or password not recognized");
+                        }
+                    } else {
+                        Log.d(TAG, "get failed with ", task.getException());
+                    }
+                });
+            } else {
+                usernameEdittext.setError("username or password empty");
+                passwordEdittext.setError("username or password empty");
+            }
+
+        });
+
+        usernameEdittext.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {}
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {}
+
+            @Override
+            public void afterTextChanged(Editable editable) {
+                String username = editable.toString().trim();
+
+                // Use Pattern and Matcher to check if the username matches the pattern
+                Pattern pattern = Pattern.compile(regex);
+                Matcher matcher = pattern.matcher(username);
+
+                if (username.isEmpty()) {
+                    usernameEdittext.setError("username cannot be empty");
+                } else if(!matcher.matches()) {
+                    usernameEdittext.setError("only alphanumeric, underscore, or period");
+                }
+            }
+        });
+
+        passwordEdittext.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {}
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {}
+
+            @Override
+            public void afterTextChanged(Editable editable) {
+                String password = editable.toString().trim();
+                if (password.isEmpty()) {
+                    passwordEdittext.setError("password cannot be empty");
+                }
+            }
+        });
+
+        signinRedirect.setOnClickListener(v -> {
+            // Replace YourSignUpFragment with the actual class name of your sign-up fragment
+            FragmentManager fragmentManager = getParentFragmentManager();
```

Comment:
this whole block does the same thing as our `navigateToFragmentPage` util, so we can replace it with that